ccm_web/server/test/app.e2e-spec.ts

```diff
@@ -1,30 +1,33 @@
-import { Test, TestingModule } from '@nestjs/testing'
-import { INestApplication, ValidationPipe } from '@nestjs/common'
 import request from 'supertest'
-import { AppModule } from './../src/app.module'
-import cookieParser from 'cookie-parser'
+import { INestApplication } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { Test, TestingModule } from '@nestjs/testing'
+
+import { AppModule } from '../src/app.module'
+import { doAppCoreSetup } from '../src/main'
+import { Config } from '../src/config'
 
 describe('APIController (e2e)', () => {
   let app: INestApplication
 
-  beforeEach(async () => {
+  beforeAll(async () => {
     const moduleFixture: TestingModule = await Test.createTestingModule({
       imports: [AppModule]
     }).compile()
 
     app = moduleFixture.createNestApplication()
-    app.use(cookieParser('SOME_COOKIE_SECRET'))
-    app.useGlobalPipes(new ValidationPipe())
-
+    const configService = app.get<ConfigService<Config, true>>(ConfigService)
+    const serverConfig = configService.get('server', { infer: true })
+    doAppCoreSetup(app, serverConfig)
```

Comment:
I think this should be OK for the current test, but at some point we'd want to modify this so that we mock or modify `Config` so that we can use a test database and make other modifications.

Reply:
Yeah... I agree